GUIDELINES.md

```diff
@@ -0,0 +1,994 @@
+
+```kotlin
+@Test
+fun `Logging in checks email at backend`() = sut {
+    givenExistingUser(USER_A)
+    whenLoggingIn(USER_A.email, USER_A.password)
+    thenEmailWasCheckedAtBackend(USER_A.email)
+}
+```
+
+The steps class is then only concerned with the technical implementation.
+
+The second aim of steps classes is to abstract not only the test code but also **configuration**. In order for the test class mostly being concerned with business-facing function calling, also the test setup should go to steps classes.
+
+An instance of a steps class can only exist once during the whole test. So if the same type of steps class is requested multiple times always the same instance is returned.
+
+Steps classes are initialized during a specific phase in the initialization of the framework and of course purged after each test function run to avoid side effects.
+
+By the way: the name "steps class" is taken from [Cucumber](https://cucumber.io/), a [behavior-driven testing](https://en.wikipedia.org/wiki/Behavior-driven_development) tool. Also there workflows are broken down into simple "steps" and grouped into steps classes. sweetest was in fact designed with interoperability with Cucumber in mind and takes the concepts further by allowing for interdependent steps classes and dependency management, where dependencies can be consumed across steps classes.
+
+#### Using steps classes
+
+To create a steps class you have to derive from the `BaseSteps` class:
+
+```kotlin
+class LoginSteps(testContext: TestContext) : BaseSteps(testContext, appModuleTestingConfiguration)
+```
+
+You have to reference the module testing configuration (in this case `appModuleTestingConfiguration`) of the module the component under test lies in.
+
+To consume a steps class you have to use the `steps` function inside a test or steps class:
+
+```kotlin
+val sut by steps<LoginSteps>()
+```
+
+By using the variable you have access to the members of the steps class. But if you don't need any access to the members and just want the steps class to load you can consider using `requireSteps` in the configuration:
+
+```kotlin
+override fun configure() = super.configure()
+    .requireSteps<LoginSteps>()
+```
+
+### Dependencies
+
+sweetest is tailored for systems where [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) is used. As you most likely have no DI during unit testing sweetest makes good for that by offering its own simple way of doing it.
+
+In sweetest all dependencies are treated as singletons, so there will be only one instance of a certain type. So it can only be used in these cases. In all other cases you have to fall back to managing object creation on your own. The automated way should cater for the very most cases, though.
+
+#### Dependency modes
+
+For each dependency type there are two possibilities: either it's configured to be mock or real. Unless you tell sweetest otherwise, all dependencies are set to the "mock" mode by default.
```

Comment:
```suggestion
For each dependency type there are two possibilities: either it's configured to be mocked or real. Unless you tell sweetest otherwise, all dependencies are set to the "mock" mode by default.
```